tests/regression/83-callstring_Stmt/09-thread_creation.c

```diff
@@ -0,0 +1,88 @@
+// PARAM: --set ana.context.callStack_height 10 --set "ana.activated[+]" callstring_stmt --enable ana.int.interval_set
+#include <pthread.h>
+#include <stdio.h>
+#include <goblint.h>
+
+int f(int i)
+{
+  int res = 0;
+  if (i == 0)
+  {
+    res = 1;
+  }
+  if (i > 0)
+  {
+    res = f(--i);
+  }
+  return res;
+}
+
+int g(int i)
+{
+  int res = 0;
+  if (i == 0)
+  {
+    res = 3;
+  }
+  if (i > 0)
+  {
+    res = g(--i);
+  }
+  return res;
+}
+
+int h(int i)
+{
+  int res = 0;
+  if (i == 0)
+  {
+    res = 2;
+  }
+  if (i > 0)
+  {
+    res = g(--i);
+  }
+  return res;
+}
+
+int procedure(int num_iterat)
+{
+  int res1 = f(num_iterat);
+  int res2 = g(num_iterat);
+  int res3 = h(num_iterat);
+  int res4 = h(num_iterat);
+  return res1 + res2 + res3 + res4;
+}
+
+void *t_sens(void *arg)
+{
+  int result = procedure(0);
+  __goblint_check(result == 8);
+
+  result = procedure(8);
+  __goblint_check(result == 10);
+  return NULL;
+}
+
+void *t_insens(void *arg)
+{
+  int result = procedure(9);
+  __goblint_check(result == 10); // UNKNOWN
+
+  result = procedure(60);
+  __goblint_check(result == 10); // UNKNOWN
+  return NULL;
+}
+
+int main()
+{
+  pthread_t id;
+  pthread_t id2;
+
+  // Creat the thread
```

Comment:
```suggestion
  // Create the thread
```

also in a few other places :wink:

Reply:
The testcases are not perfect yet :) But I will improve them in a later commit

Reply:
Sure, just wanted to remark because such things are missed easily 😉